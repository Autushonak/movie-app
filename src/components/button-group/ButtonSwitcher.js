import styled from 'styled-components';
import PropTypes from 'prop-types';
import { colors } from '../../constants/colors';

const ButtonSwitcher = styled.button`
  background-color: ${(props) => (props.active ? `${colors.PINK}` : `${colors.GREY_1}`)};
  border: none;
  ${(props) => (props.left ? 'border-radius: 4px 0 0 4px;' : '')};
  ${(props) => (props.right ? 'border-radius: 0 4px 4px 0;' : '')};
  text-transform: uppercase;
  color: ${colors.WHITE};
  text-align: center;
  font-size: 12px;
  padding: 10px 20px;
  max-height: 40px;
`;

ButtonSwitcher.propTypes = {
  title: PropTypes.string,
  active: PropTypes.bool,
  left: PropTypes.bool,
  right: PropTypes.bool,
  onClick: PropTypes.func,
};

export default ButtonSwitcher;
