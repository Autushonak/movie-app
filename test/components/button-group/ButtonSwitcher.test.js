import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import ButtonSwitcher from '../../../src/components/button-group/ButtonSwitcher';

describe('Button switcher', () => {
  test('should be rendered correctly when active left', () => {
    const component = shallow(
      <ButtonSwitcher key="title1" left={true} right={false} active={true}>
        title1
      </ButtonSwitcher>,
    );
    expect(component).toMatchSnapshot();
  });

  test('should be rendered correctly when inactive right', () => {
    const component = shallow(
      <ButtonSwitcher key="title2" left={false} right={true} active={false}>
        title2
      </ButtonSwitcher>,
    );
    expect(component).toMatchSnapshot();
  });
});
