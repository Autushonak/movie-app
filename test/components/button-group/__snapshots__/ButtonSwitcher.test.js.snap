// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button switcher should be rendered correctly when active left 1`] = `
.c0 {
  background-color: #F65261;
  border: none;
  border-radius: 4px 0 0 4px;
  text-transform: uppercase;
  color: #FFFFFF;
  text-align: center;
  font-size: 12px;
  padding: 10px 20px;
  max-height: 40px;
}

<button
  className="c0"
>
  title1
</button>
`;

exports[`Button switcher should be rendered correctly when inactive right 1`] = `
.c0 {
  background-color: #424242;
  border: none;
  border-radius: 0 4px 4px 0;
  text-transform: uppercase;
  color: #FFFFFF;
  text-align: center;
  font-size: 12px;
  padding: 10px 20px;
  max-height: 40px;
}

<button
  className="c0"
>
  title2
</button>
`;
